client.go

```diff
@@ -53,55 +74,89 @@ func NewClient(addr string, tlsConf *TLSConfig, opts ...ClientOpts) (*Client, er
 		}
 	}
 
+	if client.auth != nil {
+		if err := client.renewToken(); err != nil {
+			return nil, err
+		}
+	}
+
 	return client, nil
 }
 
-func (c *Client) Request(method string, path []string, body interface{}, parameters url.Values, response interface{}) error {
+func (c *Client) renewToken() error {
+	res, err := c.auth.Auth()
+	if err != nil {
+		return err
+	}
+
+	c.SetToken(res.Auth.ClientToken)
+
+	return nil
+}
+
+func (c *Client) Request(method string, path []string, body, response interface{}, opts *RequestOptions) error {
```

Comment:
Maybe add a `if opts == nil { opts = &RequestOptions{} }` to the top of the function? That way you can skip all the `if opts != nil` tests scattered throughout the function.

Reply:
:+1: